### PR TITLE
Add Dlartg and test

### PR DESCRIPTION
**Pull request by btracey**

### CODE REVIEW

File: native/general.go [hunk 1]
```diff
@@ -90,4 +90,6 @@ var (
 
 	smlnum = dlamchS / dlamchP
 	bignum = 1 / smlnum
+
+	dlamchB = 2
```
Comment (kortschak) [inline native/general.go]: Comment please.

File: native/dlartg.go [hunk 2]
```diff
@@ -0,0 +1,77 @@
+// Copyright ©2015 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package native
+
+import "math"
+
+// Dlartg generates a plane rotation so that
```
Comment (kortschak) [inline native/dlartg.go]: Add similar notes as at the LAPACK code that compare it to drotg.

File: native/dlartg.go [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+func (impl Implementation) Dlartg(f, g float64) (cs, sn, r float64) {
+	safmin := dlamchS
+	eps := dlamchE
+	safmn2 := math.Pow(float64(dlamchB), math.Log(safmin/eps))
+	safmx2 := 1 / safmn2
+	if g == 0 {
+		cs = 1
+		sn = 1
+		r = f
+		return
```
Comment (kortschak) [inline native/dlartg.go]: Add the return values, and below.

**Comment by kortschak** [comment]
LGTM after comments addressed.

File: native/general.go (hunk 1 above)
Comment (btracey) [inline native/general.go]: Done.

File: native/dlartg.go (hunk 2 above)
Comment (btracey) [inline native/dlartg.go]: Done.

File: native/dlartg.go (hunk 3 above)
Comment (btracey) [inline native/dlartg.go]: Done.

File: native/dlartg.go [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+// license that can be found in the LICENSE file.
+
+package native
+
+import "math"
+
+// Dlartg generates a plane rotation so that
+//  [cs  sn] * [f] = [r]
+//  [-sn cs]   [g] = [0]
+// This is a more accurate version of drotg, with the other differences that
```
Comment (kortschak) [inline native/dlartg.go]: "of BLAS drotg"

Comment (btracey) [inline native/dlartg.go]: Thanks.

**Pull request merged**